docs/contributing/mentorship.md

```diff
@@ -82,4 +82,45 @@ The example retrospective process looks as follows:
+
+* Encourage mentees who are interested in applying, by asking them to look through our issues and trying to work on one in which they would learn a bit about the codebase!
+* Allocate some time upfront for this process, as you might have a lot of mentee applications to look through.
+  * If possible, book some time on your calendars and discuss the selection process with co-mentors in a particular quarter beforehand.
+* Try to have some prerequisites in mind while shortlisting an application of a potential mentee (CV/questions/proposal/interview),
+  * The mentee should be able to allocate enough time to the mentorship.
+    * Thus, having prior important commitments like a full-time job or large educational course load can lead to mentees not having enough time for the mentorship, even if they commit to working > 30hrs a week. This could manifest as mentees not having enough energy to think about a project, skipping meetings, and generally not being able to utilize the time mentors can spend with them.
+    * So it is better to practically judge the amount of work a mentee would be able to balance before selecting them.
+    * Generally we expect our mentees to be university/college students, who would be able to balance both the mentorship and a moderate amount of course load. Commitments larger than this would need to be treated with some caution.
+  * The mentee should have at least one prior contribution to Thanos or its dependencies that affect it indirectly. The contribution should demonstrate effort & interest on part of the potential mentee.
+    * We want to understand that the potential mentee is interested in the project and has invested some time and effort in trying to understand Thanos.
+    * However, the contribution should not be a minimal effort, such as a typo/grammar fix or variable name change. While these are valued contributions, they cannot effectively demonstrate mentee effort, and should instead be some code/doc change. Note that this doesn't need to be in a merged/finalized state and can be open.
+  * The mentee should have proper means of communication (can be ensured during the 20-minute interview).
+    * This constitutes things like good enough internet to be able to video calls without dropping, as that is the primary means of communication between mentors and mentees.
+    * As most of the discussions (meetings/issues/reviews/code comments) are conducted in English, the mentee should be able to communicate well in English (first/second language).
+  * The mentee should have some form of local development environment setup, that would not require the mentor to constantly debug the mentee's environment, while working on the project.
+    * This is hard to ensure fully, but generally the mentee should be working on linux/linux VM/macOS, so most generic GNU/BSD commands are compatible.
```

Comment:
I'd even say debugging local environment is not strictly part of mentoring, ideally this should be up to the mentee and they should be able to deal with their environment (obviously, mentor should be able to follow their environment during pair programming etc.).

Reply:
I don't think this is strictly necessary and I think it's potentially exclusionary to note it. If mentees want to use windows because it's the machine they have access to, then by all means, let them. They can always use the WSL or something. Alternatively, they can help get our scripts and such working on .ore platforms.

Reply:
Yup, I can rephrase a bit here but generally wanted to mention that having a local environment that the mentee knows how to operate, is beneficial, and helps the mentorship be more productive. I think we can mention linux VM as a fallback if problems due to this arise. We're not excluding any platforms/machines by any means. WDYT? 🙂 

I think @bwplotka can speak more to this as well.

Reply:
> Unfortunately, we have a visible case where WSL blocked mentees for the majority of mentorship time.

Let's make this better then. I have a WSL2 environment at home that I successfully used for some time to work on Thanos and if there are any related issues I could have a look at them to help.

I 200% agree with @squat's statement.

Reply:
To reiterate, we aren't excluding mentees from using any machine/OS. Running a linux VM is acceptable in most cases and this is about having really productive mentorship, without the mentor having to spend time, debugging the mentee's local env in case they run into problems and are blocked. 

> Let's make this better then. I have a WSL2 environment at home that I successfully used for some time to work on Thanos and if there are any related issues I could have a look at them to help.

You could update https://thanos.io/tip/contributing/contributing.md/#development with this, but I'd still be against recommending complicated envs to mentees (who are often new to software and open-source), and just defaulting to simple linux VMs.

Reply:
It's not complicated, unless we have Service objects using `service.spec.sessionAffinity: ClientIP`. In this case we would need to recompile the WSL 2 kernel, then it becomes complicated.

Otherwise it's literally a matter of installing an app from the Microsoft Store. It integrates easily into Windows itself and any GUI code editor (like, but not limited to, VSCode or Goland).